### DISCUSSION TITLE
Hello from NestMTX

### DISCUSSION THREAD

**Discussion by jakguru** [blakeblackshear/frigate#13567] [General]
Hello. I am the maintainer of a small project called NestMTX (nestmtx.com) which is a service which is meant to help smarthome hobbyists with some technical experience access the feeds from their Google/Nest cameras for use in other services. In simple terms, it's a "proxy" which handles all of the technical aspects of retrieving a stream from Google's servers and provides simple, easy to use outputs which can be utilized in applications like Frigate.

I was hoping to get some assistance documenting the "optimal settings" for integrating a NestMTX feed into Frigate. I've already written some basic documentation on the topic (https://nestmtx.com/integrations/frigate.html) but I wanted to consult with the experts here on what is the best integration method which will provide the best performance with the least overhead.

Thanks!

**Comment by NickM-27** [comment]
Hi, what you have there looks pretty good. Users would also want to setup the camera in go2rtc so that Frigate UI uses a higher quality live view. That would just be 3 additional config lines

**Reply by jakguru** [reply]
is there a "passthrough" option to allow users to access it in the live view? since NestMTX is already doing a lot of heavy lifting and is capable of accepting as many connections as you can throw at it (well, as many connections as resources allow). Or maybe I just don't personally understand how to do it using go2rtc.

**Reply by NickM-27** [reply]
go2rtc is used at a minimum in this case to take the rtsp stream and repackage it as MSE or WebRTC stream to be used for live view. The extra config in the linked example would just be

```
go2rtc:
  streams:
    My_NestMtx_Camera:
      - rtsp://my.nestmtx.local:8554/my_nestmtx_camera
```

**Reply by jakguru** [reply]
while i can't speak for MSE, NestMTX does offer a WebRTC stream. Is that helpful / beneficial in any way to further reduce the resource load?

**Reply by NickM-27** [reply]
I don't think we are really looking to support alternative implementations, go2rtc is included in frigate for this as one of its main purposes. There would only be additional load while the user is viewing the live stream, and given the nest resolution I imagine it would be fairly negligible

**Reply by jakguru** [reply]
I completely understand. Given that all of the feeds coming out of NestMTX are H.264 Video + AAC Audio, is there a more optimal `input_args` setting than `preset-rtsp-restream`? Is there anything else that you think I can add to the tutorial which could contribute to a user's ease of setup?

**Reply by NickM-27** [reply]
no, I think the restream input is the right choice

**Comment by Gooshy** [comment]
Hi,
I've got a new setup of 2 Nest cameras up and live in NestMTX.  I have added them in Frigate and have live feeds.  Every 3-5 minutes I lose connection and get a blue screen shown below.  Where's the best place to debug?

![Screenshot 2025-02-14 000412](https://github.com/user-attachments/assets/46a54f0e-0e34-42f8-a62f-57faf8f40002)

**Comment by Waggers58** [comment]
I am following the documentation to setup Nest MTX but I cannot get the Google Cloud Platform console to accept the displayed Redirect URL as an Authorised Redirect URL.

I enter 'https://192.168.0.85:2001/credentials/authorize' as suggested, Google returns:

Invalid redirect: Must end with a public top-level domain (such as .com or .org).
Invalid Redirect: must use a domain that is a valid [Top private domain ]

I can get past this by entering 'https://localhost/credentials/authorize' but then, when I attempt to authorize my credentials', Google returns:

Access blocked: Authorization Error
device_id and device_name are required for private IP: https://192.168.0.85:2001/credentials/authorize

I am running version 1.1.1 of NestMTX, the AMD64 version, in Docker under Linux Mint 21.1 Cinnamon (but that is probably irrelevant). I assume Google have been changing all their routines again.

Any advice?

**Reply by mikespas123** [reply]
they dont allow local network IPs so I used tailscale to mask it

Example NestPass URL:

http://homework.tail50fee3.ts.net:2000

Matching Google OAuth Redirect URI:

http://homework.tail50fee3.ts.net:2000/auth/callback
